conclave-common/src/main/kotlin/com/r3/conclave/common/internal/StreamCallInterfaceMessage.kt

```diff
@@ -0,0 +1,58 @@
+package com.r3.conclave.common.internal
+
+import com.r3.conclave.mail.internal.readInt
+import com.r3.conclave.mail.internal.writeInt
+import com.r3.conclave.utilities.internal.*
+import java.io.InputStream
+import java.io.OutputStream
+import java.nio.ByteBuffer
+
+class StreamCallInterfaceMessage(
+        val hostThreadID: Long,
+        val callTypeID: Byte,
+        val messageTypeID: Byte,
+        val payload: ByteArray?
+) {
+    private val headerSize = 8 + 1 + 1
+    private val payloadSize get() = (payload?.size ?: 0) + 1
+
+    fun size() = headerSize + payloadSize
+
+    companion object {
+        fun fromBytes(bytes: ByteArray): StreamCallInterfaceMessage {
+            val buffer = ByteBuffer.wrap(bytes)
+
+            val hostThreadID = buffer.long
+            val callTypeID = buffer.get()
+            val messageTypeID = buffer.get()
+            val messageBytes = buffer.getNullable { getRemainingBytes() }
```

Comment:
Just conscious here that we might end up doing extra copying, all to wrap the payload into a message object.

Reply:
I didn't put much thought into the serialisation of the message data object! Will remove this in favour of `Data(Input|Output)Stream`.

Reply:
I tried this and it's actually substantially slower! (at least in the hammer tests that I've written)